test/setup-ca.sh

```diff
@@ -18,8 +18,12 @@ fi
 
 CA_CRT=$(realpath ${CA}.pem)
 CA_KEY=$(realpath ${CA}-key.pem)
-if ! [ -f ${CA_CRT} -a -f ${CA_KEY} ]; then
-  echo "Generating CA certificate ..."
+CADIR=$(dirname ${CA})
+if ! [ -d ${CADIR} -a -f ${CA_CRT} -a -f ${CA_KEY} ]; then
+  dir=$PWD
+  mkdir -p $CADIR
+  cd "$CADIR"
+  echo "Generating CA certificate in $CADIR..."
```

Comment:
Instead of calling `cd` multiple times and having to remember the original directory, can you do `(cd <domething> && <some command>)`, i.e. change the directory in a sub-shell?

Reply:
It's a more localized change and less likely to cause maintenance issues when we change the code and forget in which directory we are.